src/js/components/Tab/Tab.js

```diff
@@ -175,23 +203,23 @@ const Tab = forwardRef(
         onMouseOver={onMouseOverTab}
         onMouseOut={onMouseOutTab}
         onFocus={() => {
-          setFocus(true);
-          if (onMouseOver) onMouseOver();
+          if (onFocus) onFocus();
+          setFocusIndex(index);
         }}
         onBlur={() => {
-          setFocus(undefined);
-          if (onMouseOut) onMouseOut();
+          if (onBlur) onBlur();
+          setFocusIndex(-1);
         }}
-        // ensure focus outline is not covered by hover styling
-        // of adjacent tabs
-        style={focus && { zIndex: 1 }}
       >
         <StyledTab
           as={Box}
           disabled={disabled}
           plain={plain}
           {...withIconStyles}
           {...tabStyles}
+          style={{
```

Comment:
We should put this inside StyledTab itself, instead of using the `style` property here.

Reply:
Fixed in latest commit